doc/19-technical-concepts.md

```diff
@@ -1938,6 +1938,68 @@ Message updates will be dropped when:
 * Checkable does not exist.
 * Origin endpoint's zone is not allowed to access this checkable.
 
+#### event::SetCommentRemovalInfo <a id="technical-concepts-json-rpc-messages-event-setcommentremovalinfo"></a>
+
+> Location: `clusterevents.cpp`
+
+##### Message Body
+
+Key       | Value
+----------|---------
+jsonrpc   | 2.0
+method    | event::SetCommentRemovalInfo
+params    | Dictionary
+
+##### Params
+
+Key            | Type        | Description
+---------------|-------------|---------------------------------
+comment        | String      | Comment name
+removed\_by    | String      | Name of the removal requestor
+remove\_time   | Timestamp   | Time of the remove operation
+
+##### Functions
+
+Event Sender: `Comment::OnRemovalInfoChanged`
+Event Receiver: `SetCommentRemovalInfoAPIHandler`
+
+This message is used to synchronize information about manual comment removal before deleting the corresponding object.
+
+##### Permissions
+
+This message is only accepted from the local zone and from parent zones.
+
+#### event::SetDowntimeCancellationInfo <a id="technical-concepts-json-rpc-messages-event-setdowntimecancellationinfo"></a>
+
+> Location: `clusterevents.cpp`
+
+##### Message Body
+
+Key       | Value
+----------|---------
+jsonrpc   | 2.0
+method    | event::SetDowntimeCancellationInfo
+params    | Dictionary
+
+##### Params
+
+Key            | Type        | Description
+---------------|-------------|---------------------------------
+comment        | String      | Comment name
+removed\_by    | String      | Name of the removal requestor
+cancel\_time   | Timestamp   | Time of the cancel operation
```

Comment:
IMAO a non-equal, non-consistent wording: remove <> cancel. While unifying I'd prefer remove to match the comment counterpart.

Reply:
I thought "cancel" was the more frequently used terminology for downtimes, so I went with that for everything except where different naming already existed. But that seems to be more of an Icinga DB thing, the Icinga 2 API action is called "remove-downtime" after all. I think I could also just unify both into a single `event::SetRemovalInfo` cluster message.

Reply:
... which – if you do – shall be type-aware btw.

Reply:
What do you mean by that?

Reply:
Can you tell by object name wether it's a comment/downtime?

Reply:
No